core/src/db/engine/ExecutionEngineImpl.cpp

```diff
@@ -636,6 +640,34 @@ ExecutionEngineImpl::CopyToFpga() {
     return Status::OK();
 }
 
+Status
+ExecutionEngineImpl::CopyToApu(uint32_t row_count) {
+#ifdef MILVUS_APU_VERSION
+
+    auto cache_index_ =
+        std::static_pointer_cast<knowhere::VecIndex>(cache::FpgaCacheMgr::GetInstance()->GetItem(location_));
+    bool already_in_cache = (cache_index_ != nullptr);
+
+    if (!already_in_cache) {
+        cache::FpgaCacheMgr::GetInstance()->ClearCache();  // clear cache to support cache switch .
```

Comment:
It seems that you want to cache only one segment index, isn't it?

Reply:
Hi , yes . 
Currntly the use of FpgaCachch is to hold the latest data set loaded to Apu according location .we are supporting load of 1 data set at a time so no need to hold more then one index. 
other funcionality is been managed by Apu libs.